internal/checker/cloudformation/rules/general-security/verify_no_access_backend_api.rego

```diff
@@ -0,0 +1,50 @@
+# METADATA
+# title: "Verify no open access to back-end through API"
+# description: "This policy underscores that AWS API Gateway methods are initially public. To safeguard against data leaks and security breaches, all methods in the API Gateway should be protected by an Authorizer or an API key. It's recommended to configure a custom authorizer or an API key for each method to bolster security."
+# scope: package
+# related_resources:
+# - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-method.html
+# custom:
+#   id: CB_CFT_46
+#   severity: LOW
+package lib.cloudformation.CB_CFT_46
+
+import future.keywords.in
+
+resource := "AWS::ApiGateway::Method"
+
+is_valid {
+	some resources in input.Resources
+	resources.Type == resource
+}
+
```

Comment:
swap logic for pass and fail
fail[properties] {
	is_valid
	some resources in input.Resources
	properties := resources.Properties
	properties.HttpMethod != "OPTIONS"
	properties.AuthorizationType == "NONE"
	properties.ApiKeyRequired == false
}